features/adresseerbaarobject-embedden-in-adres.feature

```diff
@@ -0,0 +1,68 @@
+# language: nl
+Functionaliteit: Adresseerbaar object embedden in een adres
+
+	Als consumer van de BAG API
+	Wil ik bij het opvragen van Adressen tegelijk het bijbehorende Adresseerbaar Object kunnen opvragen
+	Zodat ik niet een aparte aanroep hoeft te doen voor het opvragen van het bijbehorende Adresseerbaar object
+
+Abstract Scenario: Opvragen van adressen en bijbehorend adresseerbaar object via het /adressen endpoint
+    Als GET /adressen<query part>expand=adresseerbaarObject is aangeroepen
+    Dan bevat de _embedded.adresseerbaarObject property van elk Adres in de response het bijbehorend Adresseerbaar Object
+
+    Voorbeelden:
+        | query part                                                        |
+        | ?pandIdentificatie=0193100000048288&                              |
+        | ?adresseerbaarObjectIdentificatie=0484010002033603&               |
+        | ?zoekresultaatIdentificatie=adr-89d5a4d96f09c60716c4671fdb9334b8& |
+        | ?postcode=2391PH&huisnummer=1&                                    |
+        | /0826200000016921?                                                |
+
+Abstract Scenario: Opvragen van adressen en een deel van de kenmerken van bijbehorend adresseerbaar object via het /adressen endpoint
+    Als GET /adressen<query part>expand=adresseerbaarObject.gebruiksdoelen,adresseerbaarObject.geconstateerd,adresseerbaarObject.geometrie is aangeroepen
+    Dan bevat de _embedded.adresseerbaarObject property van elk Adres in de response de gevraagde kenmerken van het bijbehorend Adresseerbaar Object
+
+    Voorbeelden:
+        | query part                                                        |
+        | ?pandIdentificatie=0193100000048288&                              |
+        | ?adresseerbaarObjectIdentificatie=0484010002033603&               |
+        | ?zoekresultaatIdentificatie=adr-89d5a4d96f09c60716c4671fdb9334b8& |
+        | ?postcode=2391PH&huisnummer=1&                                    |
+        | /0826200000016921?                                                |
+
+Scenario: Opvragen van adressen en bijbehorend adresseerbaar object met overtollig expand waarden
+    Als GET /adressen?pandIdentificatie=0193100000048288&expand=<expand> is aangeroepen
+    Dan bevat de _embedded.adresseerbaarObject property van elk Adres in de response de gevraagde kenmerken van het bijbehorend Adresseerbaar Object
+
+    Voorbeelden:
+        | expand                                                                |
+        | adresseerbaarObject,adresseerbaarObject                               |

```

Comment:
Deze begrijp ik niet, waarom 2x. Moet dit niet gewoon

| adresseerbaarObject

zijn?

Reply:
Dat klopt. Dit kan een fout zijn die een consumer maakt en dat geldt ook voor de 2e voorbeeld. Het gedrag bij de andere embedded objects is dat het volledig embedded object wordt geretourneerd. Om er zeker van te zijn dat dit gedrag ook voor adresseerbaarObject hetzelfde is en dat het niet wordt vergeten tijdens het acceptatie testen, heb ik dit als scenario toegevoegd

Reply:
Aha, dit is dus eigenlijk een scenario waarmee duidelijk wordt hoe om te gaan met foute invoer van een Client.
Ik weet niet of dat kan maar zou je hier dan niet eigenlijk ook een melding willen sturen op basis waarvan de ontwikkelaars van de Client applicatie deze fout kunnen herstellen?

Reply:
Het kan, maar dat betekent dat je dit ook voor de andere embedded resources moet wijzigen en dan is het een breaking change.